editor/plugins/particle_process_material_editor_plugin.cpp

```diff
@@ -0,0 +1,450 @@
+					range_edit_widget->queue_redraw();
+				}
+			} break;
+
+			case DRAG_LEFT:
+			case DRAG_RIGHT: {
+				float new_value = drag_from_value + (mm->get_position().x - drag_origin) / usable_area.x * property_range;
+				if (drag == DRAG_LEFT) {
+					new_value = MIN(new_value, max_range->get_value());
+					_set_clamped_values(new_value, max_range->get_value());
+				} else {
+					new_value = MAX(new_value, min_range->get_value());
+					_set_clamped_values(min_range->get_value(), new_value);
+				}
+			} break;
+
+			case DRAG_MIDDLE: {
+				float delta = (mm->get_position().x - drag_origin) / usable_area.x * property_range;
+				float diff = max_range->get_value() - min_range->get_value();
+				delta = CLAMP(drag_from_value + delta, min_range->get_min(), max_range->get_max() - diff) - drag_from_value;
+				_set_clamped_values(drag_from_value + delta, drag_from_value + delta + diff);
+			} break;
+
+			case DRAG_SCALE: {
+				float delta = (mm->get_position().x - drag_origin) / usable_area.x * property_range + drag_from_value;
+				_set_clamped_values(MIN(drag_pivot, drag_pivot - delta), MAX(drag_pivot, drag_pivot + delta));
+			} break;
+		}
+	}
+}
+
+void ParticleProcessMaterialMinMaxPropertyEditor::_set_mouse_inside(bool p_inside) {
+	mouse_inside = p_inside;
+	if (!p_inside && drag == DRAG_NONE) {
+		hover = HOVER_NONE;
+	}
+	range_edit_widget->queue_redraw();
+}
+
+float ParticleProcessMaterialMinMaxPropertyEditor::_get_left_offset() const {
+	return margin.x + usable_area.x * min_range->get_as_ratio();
+}
+
+float ParticleProcessMaterialMinMaxPropertyEditor::_get_right_offset() const {
+	return margin.x + usable_area.x * max_range->get_as_ratio();
+}
+
+Rect2 ParticleProcessMaterialMinMaxPropertyEditor::_get_middle_rect() const {
+	if (Math::is_equal_approx(min_range->get_value(), max_range->get_value())) {
+		return Rect2();
+	}
+
+	return Rect2(
+			Vector2(_get_left_offset() - 1, margin.y),
+			Vector2(usable_area.x * (max_range->get_as_ratio() - min_range->get_as_ratio()) + 1, usable_area.y));
+}
+
+void ParticleProcessMaterialMinMaxPropertyEditor::_set_clamped_values(float p_min, float p_max) {
+	// This is required for editing widget in case the properties have or_less or or_greater hint.
+	min_range->set_value(MAX(p_min, min_range->get_min()));
+	max_range->set_value(MIN(p_max, max_range->get_max()));
+	_update_slider_values();
+	_sync_property();
+}
+
+void ParticleProcessMaterialMinMaxPropertyEditor::_sync_property() {
+	const Vector2 value = Vector2(min_range->get_value(), max_range->get_value());
+	emit_changed(get_edited_property(), value, "", true);
+	range_edit_widget->queue_redraw();
+}
+
+void ParticleProcessMaterialMinMaxPropertyEditor::_update_mode() {
+	switch (slider_mode) {
+		case MODE_RANGE: {
+			// TRANSLATORS: Short for Minimum. This string should be very short.
+			min_edit->set_label(TTR("min"));
+			// TRANSLATORS: Short for Maximum. This string should be very short.
+			max_edit->set_label(TTR("max"));
```

Comment:
I'm thinking that we shouldn't make these able to be localized. What do you think?

Reply:
You mean only min/max or all 4?
I think piv/dev might be worth translating, because they are not that obvious, and min/max mostly for consistency (they are not the same in every language too).

Reply:
3-4 letters for every language to try and convey these concepts seems absurd to me. It's frustrating because I think everything else about the UI and its sizing looks perfect to me now...

Reply:
I'll remove it *for now*.